src/NuGet.Core/NuGet.Build.Tasks/BuildTasksUtility.cs

```diff
@@ -272,6 +272,27 @@ public static void AddPropertyIfExists(IDictionary<string, string> properties, s
             }
         }
 
+        /// <summary>
+        /// Try to parse the <paramref name="restoreProjectStyleProperty"/> and return the <see cref="ProjectStyle"/> value.
+        /// </summary>
+        /// <param name="restoreProjectStyleProperty">The value of the RestoreProjectStyle property value. It can be null.</param>
+        /// <returns>The <see cref="ProjectStyle"/>. If the <paramref name="restoreProjectStyleProperty"/> is null the return vale will be null. </returns>
+        public static ProjectStyle? GetProjectRestoreStyleFromProjectProperty(string restoreProjectStyleProperty)
```

Comment:
I introduced this call to avoid reading the MsBuildItem properties multiple times. The method GetProjectRestoreStyle has more logic in addition to property reading. The data is read once and used in multiple calls after.